run_terminate_imagebuilder_gendo_old_environment.py

```diff
@@ -99,33 +102,68 @@ def run_terminate_image(name):
-
-    timestamp_list = list()
-    for cc in component_list:
-        try:
-            found = re.search('-component-(.+?)/', cc).group(1)
-            if found:
-                timestamp_list.append(found)
-        except AttributeError:
-            pass
-    timestamp_list = set(timestamp_list)
+    print_message('check in used ami version')
+    ec2_describe_role_arn = ''
+    for settings in env.get('imagebuilder', list()):
+        if settings['NAME'] == 'run_terminate_imagebuilder_gendo_old_environment':
+            ec2_describe_role_arn = settings['EC2_DESCRIBE_ROLE_ARN']
+
+    if ec2_describe_role_arn:
+        cmd = ['sts', 'assume-role']
+        cmd += ['--role-arn', ec2_describe_role_arn]
+        cmd += ['--role-session-name', 'ec2-describe-role']
+        rr = aws_cli.run(cmd)
+
+        access_key = rr['Credentials']['AccessKeyId']
+        secret_key = rr['Credentials']['SecretAccessKey']
+        session_token = rr['Credentials']['SessionToken']
+
+        aws_cli_for_ec2 = AWSCli(aws_access_key=access_key,
+                                 aws_secret_access_key=secret_key,
+                                 aws_session_token=session_token)
+
+        cmd = ['elasticbeanstalk', 'describe-environments']
+        rr = aws_cli_for_ec2.run(cmd)
+
+        gendo_eb_name_list = list()
+        for r in rr['Environments']:
+            if r['EnvironmentName'].startswith('gendo'):
+                gendo_eb_name_list.append((r['EnvironmentName']))
+
+        in_use_ec2_ami_list = list()
+        for eb_name in gendo_eb_name_list:
+            cmd = ['ec2', 'describe-instances']
+            cmd += ['--filters', f'Name=tag:elasticbeanstalk:environment-name,Values={eb_name}']
+            rr = aws_cli_for_ec2.run(cmd)
+            if rr['Reservations'] and rr['Reservations'][0]['Instances']:
+                in_use_ec2_ami_list.append(rr['Reservations'][0]['Instances'][0]['ImageId'])
 
     print_message('get All imagebuilder resouce version')
-    cmd = ['imagebuilder', 'list-images']
-    rr = aws_cli.run(cmd)
-    ami_arn_list = filter_imagebuilder_resource_arn_list(rr['imageVersionList'])
+
+    print_message('describe log groups imagebuilder resouce')
+    cmd = ['logs', 'describe-log-groups']
+    cmd += ['--log-group-name-prefix', '/aws/imagebuilder/gendo']
+    rr = aws_cli.run(cmd, ignore_error=True)
+    imagebuilder_cw_log_list = rr['logGroups']
 
     cmd = ['ec2', 'describe-images']
     cmd += ['--filters=Name=name,Values="Gendo*"']
     cmd += ['--owners', account_id]
     rr = aws_cli.run(cmd)
     ec2_gendo_img_list = rr['Images']
 
+    in_use_ami_timestamp_version = list()
+    for img in ec2_gendo_img_list:
+        if img['ImageId'] in in_use_ec2_ami_list:
+            for tag in img['Tags']:
+                if tag['Key'] == 'Ec2ImageBuilderArn':
+                    m = re.search('/gendo-recipe-(.+?)/', tag['Value'])
+                    in_use_ami_timestamp_version.append(m.group(1))
```

Comment:
@GoldieKMG // if 조건문이 단순하다면, 약간 변경하면 들여쓰기를 1~2depth 줄여서 가독성을 높일수 있습니다. 이런 스타일의 작성을 부탁드립니다.
```suggestion
    for img in ec2_gendo_img_list:
        if img['ImageId'] not in in_use_ec2_ami_list:
           continue
        for tag in img['Tags']:
              if tag['Key'] != 'Ec2ImageBuilderArn':
                  continue
              m = re.search('/gendo-recipe-(.+?)/', tag['Value'])
              in_use_ami_timestamp_version.append(m.group(1))
```

Reply:
@yoonjs2 // 뎁스를 최대한 안나오게 하고 싶었는데 이렇게 풀면 됐었네요 감사합니다 :) 다음 작업 시 참고하겠습니다 👍